core/src/main/clojure/xtdb/types.clj

```diff
@@ -805,6 +805,10 @@
   ^long [^long length ^long typlen]
   (+ 4 (* 4 4) (* typlen length) (* 4 length)))
 
+(defn calculate-variable-size-array-buffer-size
+  ^long [^long element-count ^long payload-size]
+  (+ 4 (* 4 4) payload-size (* 4 element-count)))
```

Comment:
subjective: I'd inline this so that it's closer to the knowledge of what these 4s represent